packages/sdk/src/conversation/sso/ssoExecutionDialog.ts

```diff
@@ -0,0 +1,307 @@
+    if (dialogName) {
+      DIALOG_NAME = dialogName;
+      TEAMS_SSO_PROMPT_ID = dialogName + TEAMS_SSO_PROMPT_ID;
+      COMMAND_ROUTE_DIALOG = dialogName + COMMAND_ROUTE_DIALOG;
+    }
+
+    this.initialDialogId = COMMAND_ROUTE_DIALOG;
+
+    this.dedupStorage = dedupStorage;
+    this.dedupStorageKeys = [];
+
+    const ssoDialog = new TeamsBotSsoPrompt(teamsfx, TEAMS_SSO_PROMPT_ID, ssoPromptSettings);
+    this.addDialog(ssoDialog);
+
+    const commandRouteDialog = new WaterfallDialog(COMMAND_ROUTE_DIALOG, [
+      this.commandRouteStep.bind(this),
+    ]);
+    this.addDialog(commandRouteDialog);
+  }
+
+  /**
+   * Add TeamsFxBotSsoCommandHandler instance
+   * @param handler {@link SsoExecutionDialogHandler} callback function
+   * @param triggerPatterns The trigger pattern
+   */
+  public addCommand(handler: SsoExecutionDialogHandler, triggerPatterns: TriggerPatterns): void {
+    const commandId = uuidv4();
+    const dialog = new WaterfallDialog(commandId, [
+      this.ssoStep.bind(this),
+      this.dedupStep.bind(this),
+      async (stepContext: any) => {
+        const tokenResponse: TeamsBotSsoPromptTokenResponse = stepContext.result.tokenResponse;
+        const context: TurnContext = stepContext.context;
+        const message: CommandMessage = stepContext.result.message;
+
+        try {
+          if (tokenResponse) {
+            handler(context, tokenResponse, message);
+          } else {
+            await context.sendActivity("Failed to retrieve user token from conversation context.");
+          }
+          return await stepContext.endDialog();
+        } catch (error: unknown) {
+          await context.sendActivity("Failed to retrieve user token from conversation context.");
+          await context.sendActivity((error as Error).message as string);
+          return await stepContext.endDialog();
+        }
+      },
+    ]);
+
+    this.commandMapping.set(commandId, triggerPatterns);
+    this.addDialog(dialog);
+  }
+
+  /**
+   * The run method handles the incoming activity (in the form of a DialogContext) and passes it through the dialog system.
+   *
+   * @param context The context object for the current turn.
+   * @param accessor The instance of StatePropertyAccessor for dialog system.
+   */
+  public async run(context: TurnContext, accessor: StatePropertyAccessor) {
+    const dialogSet = new DialogSet(accessor);
+    dialogSet.add(this);
+
+    const dialogContext = await dialogSet.createContext(context);
+    this.ensureMsTeamsChannel(dialogContext);
+    const results = await dialogContext.continueDialog();
+    if (results && results.status === DialogTurnStatus.empty) {
+      await dialogContext.beginDialog(this.id);
+    }
+  }
+
+  private getActivityText(activity: Activity): string {
+    let text = activity.text;
+    const removedMentionText = TurnContext.removeRecipientMention(activity);
+    if (removedMentionText) {
+      text = removedMentionText
+        .toLowerCase()
```

Comment:
minor: may put this logic to the beginning?

Reply:
already removed